### DISCUSSION TITLE
Modify Layout of ImportAction mapping modal

### DISCUSSION THREAD

**Discussion by emjay64** [filamentphp/filament#14064] [Help]
### Package

Actions

### Package Version

3.2

### How can we help you?

Is it possible to modify the layout of the mapping selects on the ImportAction modal , with columns(), Grid or Fieldset etc...

**Comment by clementmas** [comment]
It doesn't look easily feasible with the current implementation.

You'd need to override the whole `setUp()` method of the trait `Filament\Actions\Concerns\CanImportRecords.php` and duplicate a lot of logic.

I think the form components and action logic should be extracted into smaller functions that could be overriden. You could PR that.

I personally wanted to flip the mapping to go from CSV fields to Import columns but that also requires a lot of changes.
